### PR TITLE
Update workflow continuous integration

### PR DESCRIPTION
**Pull request by hectorerb**
### Changes description

- Use references to handle jobs
- Generate docs, coverage and test report independently
- Update fastlane dependency
- Update CircleCI workflow

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@hectorerb |32|

### References

Closes #N/A
Related https://github.com/flyve-mdm/ios-mdm-dashboard/issues/52
Depends on #N/A

### CODE REVIEW

**Review by ajsb85 — DISMISSED** [review]

File: ci/scripts/screenshots.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#!/bin/sh
+
+# LICENSE
+#
+# screenshots.sh is part of FlyveMDMAdminDashboard
```
Comment (rafaelje) [inline ci/scripts/screenshots.sh]: Review the new @Naylin15 format for headers, now the name of the file is not present to be more flexible and readable

**Review by rafaelje — DISMISSED** [review]

**Comment by rafaelje** [comment]
@Naylin15 please review the documentation script

**Comment by Naylin15** [comment]
Sure 👍

**Comment by Naylin15** [comment]
Hi @hectorerb 

Here is the format for the .sh license headers [Hackmd](https://hackmd.io/jgICKmVOTxuioURFHQo0dA?view)

**Review by Naylin15 — APPROVED** [review]

**Comment by hectorerb** [comment]
Hi @Naylin15 
Thanks, license updated

**Review by rafaelje — APPROVED** [review]

**Pull request merged**